src/libp2p/peers.rs

```diff
@@ -1171,6 +1171,20 @@ where
         self.inner.respond_in_request(id.0, response)
     }
 
+    /// Returns `true` if there exists an established connection with the given peer.
+    pub fn has_established_connection(&self, peer: &PeerId) -> bool {
+        let peer_index = match self.peer_indices.get(peer) {
+            Some(idx) => *idx,
+            None => return false,
+        };
+
+        self.connections_by_peer
```

Comment:
why would we want to have multiple connections to the same peer?

Reply:
This can happen accidentally if two peers connect to each other at the same time.

Another situation is if you connect to an IP address and expect to find peer A but actually connect to peer B, even though you're already connected to B.
In that case, from the point of view of peer B, maybe you have accidentally connected to it (the example that I just gave), but maybe the first connection you had with B is actually dead and you're reopening a new one.

The best way to handle these situations is to allow multiple connections between peers, and automatically close unused ones.